docs/java-rest/high-level/apis/search.asciidoc

```diff
@@ -1,4 +1,119 @@
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/SearchDocumentationIT.java[search-request-basic]
+--------------------------------------------------
+
+<1> Creates the `SeachRequest`. Without arguments this runs against all indices.
+<2> Most parameters of the search can be added to the `SearchSourceBuilder` which contains everything that
+in the Rest API would be placed in the search request body.
+<3> Add a `match_all` query to the `SearchSourceBuilder`.
+
+==== Optional request arguments
+
+Lets first look at some of the optional argument of a `SearchRequest`.
+First of all, the request can be restricted to one or more indices or types in the following way:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/SearchDocumentationIT.java[search-request-indices-types]
+--------------------------------------------------
+
+Also, if necessary, a routing parameter can be set:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/SearchDocumentationIT.java[search-request-routing]
+--------------------------------------------------
+
+By setting `IndicesOptions`, you can control how unavailable indices are resolved and how wildcard expressions are expanded:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/SearchDocumentationIT.java[search-request-indicesOptions]
+--------------------------------------------------
+
+The preference parameter can e.g. be used to execute the search to prefer local shards. This defaults to randomize across shards.
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/SearchDocumentationIT.java[search-request-preference]
+--------------------------------------------------
+
+==== Using the SearchSourceBuilder
+
+Most options controlling the search behavior can be set on the `SearchSourceBuilder`,
+which contains more or less the equivalent of the options in the search request body of the Rest API.
+
+Here are a few examples of some common options:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/SearchDocumentationIT.java[search-source-basics]
+--------------------------------------------------
+<1> Create a `SearchSourceBuilder` with default options.
+<2> Set the query. Can be any type of `QueryBuilder`
+<3> Set the `from` option that determines the result index to start searching from. Defaults to 0.
+<4> Set the `size` option that determines the number of search hits to return. Defaults to 10.
+<5> Set an optional timeout that controls how long the search is allowed to take.
+
+==== SearchResponse
```

Comment:
between request and response, can you add the Synchronous execution and Asynchronous execution like in the other doc pages?